src/actions/collection.ts

```diff
@@ -0,0 +1,75 @@
+import { Action } from '@ngrx/store';
+import { Book } from '../models/book';
+import { label } from '../util';
+
+
+export const CollectionActionTypes = {
+  ADD_BOOK: label('[Collection] Add Book'),
+  ADD_BOOK_SUCCESS: label('[Collection] Add Book Success'),
+  ADD_TO_COLLECTION_FAIL: label('[Collection] Add Book Fail'),
+  REMOVE_BOOK: label('[Collection] Remove Book'),
+  REMOVE_BOOK_SUCCESS: label('[Collection] Remove Book Success'),
+  REMOVE_BOOK_FAIL: label('[Collection] Remove Book Fail'),
+  LOAD_COLLECTION: label('[Collection] Load Collection'),
+  LOAD_COLLECTION_SUCCESS: label('[Collection] Load Collection Success')
+};
+
+
+export class AddBook implements Action {
```

Comment:
The only way to get the tagged union types to work in TypeScript 2 is to have an interface that defines the shape of each action tagged with a string literal:

```ts
const ADD_BOOK: '[Book] Add' = '[Book] Add';

interface AddBookAction {
  type: typeof ADD_BOOK;
  payload: Book;
}

function addBook(book: Book): AddBookAction {
  return {
    type: ADD_BOOK,
    payload: book
  };
}
```

The `label` utility and using classes are the best ways I identified to reduce some of this boilerplate.